sonic-utilities-tests/aclshow_test.py

```diff
@@ -0,0 +1,194 @@
+RULE_9        DATAACL                     0              0
+RULE_6        DATAACL                     0              0
+RULE_08       DATAACL                     0              0
+DEFAULT_RULE  DATAACL                     0              0
+RULE_4        DATAACL                     0              0
+"""
+
+clear_output = ''
+
+class Aclshow():
+    def __init__(self, *args, **kwargs):
+        """
+        nullify_on_start, nullify_on_exit will call nullify_counters()
+        before and/or after the test. By default - clear on start and exit.
+        """
+        self.nullify_on_start, self.nullify_on_exit = args if args else (True, True)
+        self.kwargs = kwargs
+        self.setUp()
+        self.runTest()
+        self.tearDown()
+
+    def nullify_counters(self):
+        """
+        This method is used to empty dumped counters
+        if exist in /tmp/.counters_acl.p (by default).
+        """
+        if os.path.isfile(COUNTER_POSITION):
+            with open(COUNTER_POSITION, 'wb') as fp:
+                json.dump([], fp)
+
+    def runTest(self):
+        """
+        This method invokes main() from aclshow utility (parametrized by argparse)
+        parametrized by mock argparse.
+        """
+        @mock.patch('argparse.ArgumentParser.parse_args', return_value = argparse.Namespace(**self.kwargs))
+        def run(mock_args):
+            main()
+        run()
+
+    def setUp(self):
+        if self.nullify_on_start:
+            self.nullify_counters()
+        self.old_stdout = sys.stdout
+        self.result = StringIO()
+        sys.stdout = self.result
+
+    def tearDown(self):
+        if self.nullify_on_exit:
+            self.nullify_counters()
+        sys.stdout = self.old_stdout
+        #print self.result.getvalue()   # Print output for debugging.
+
+def test_default():
+    test = Aclshow(all = None, clear = None, rules = None, tables = None, verbose = None)
+    assert test.result.getvalue() == default_output
+
+def test_all():
+    test = Aclshow(all = True, clear = None, rules = None, tables = None, verbose = None)
+    assert test.result.getvalue() == all_output
+
+def test_rule1_dataacl():
+    test = Aclshow(all = None, clear = None, rules = 'RULE_1', tables = 'DATAACL', verbose = None)
+    assert test.result.getvalue() == rule1_dataacl_output
+
+def test_rule05_all():
+    test = Aclshow(all = True, clear = None, rules = 'RULE_05', tables = None, verbose = None)
+    assert test.result.getvalue() == rule5_all_output
+
+def test_rule0():
+    test = Aclshow(all = None, clear = None, rules = 'RULE_0', tables = None, verbose = None)
+    assert test.result.getvalue() == rule0_output
+
+def test_rule4_rule6_verbose():
+    test = Aclshow(all = None, clear = None, rules = 'RULE_4,RULE_6', tables = None, verbose = True)
+    assert test.result.getvalue() == rule4_rule6_verbose_output
+
+def test_everflow():
+    test = Aclshow(all=None, clear=None, rules=None, tables='EVERFLOW', verbose=None)
```

Comment:
if mirroring was configured, were we suppose to see the data? if so, i think we should have dedicated case for it as well. i see below the table configured but are we missing additional configuration to see the info?

Reply:
Added two rules to be in Everflow.